### DISCUSSION TITLE
Add file length and modification time to split

### DISCUSSION THREAD

**Discussion by acvictor** [facebookincubator/velox#9305] [Ideas]
In Spark, [file length and modification time is present in PartitionedFile](https://github.com/apache/spark/blob/63529bf97ad0bb6d46f4d88bc1425cad76add1a1/sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/FileScanRDD.scala#L60) although [only file name, offset, and length is used to construct the split which is then sent downstream from Gluten to Velox.](https://github.com/apache/incubator-gluten/blob/3b9a3ea093e64e6b4a3b208dc7f805292ecf117b/backends-velox/src/main/scala/io/glutenproject/backendsapi/velox/IteratorApiImpl.scala#L98)

At present, an additional call is made in Velox to fetch metadata and initialize file length although the information is present upstream and not used to construct the split. For example in ABFS https://github.com/facebookincubator/velox/blob/90fc393715b70e78bf8bf7f4a14272c50b49030a/velox/connectors/hive/storage_adapters/abfs/AbfsFileSystem.cpp#L68 and in S3 https://github.com/facebookincubator/velox/blob/90fc393715b70e78bf8bf7f4a14272c50b49030a/velox/connectors/hive/storage_adapters/s3fs/S3FileSystem.cpp#L92 

If fileSize and modificationTime from every PartitionedFile object are added to HiveConnectorSplit we can save on one network call per file handle. Additionally, if these are appended to the key used in the file handle cache, then the cache can be enabled by default even when files are being overwritten because the modification time and length will be different. If this idea sounds good I would like to take this up. We've seen improvement in the order of 10<sup>2</sup> seconds in total scan time across all tasks for TPCDS and a modest gain overall.

@mbasmanova @zhli1142015

**Comment by mbasmanova** [comment]
Sounds interesting. @majetideepak Deepak, what do you think?

**Comment by majetideepak** [comment]
This sounds reasonable to me. I can see the benefit if there are many small files.

**Reply by majetideepak** [reply]
We should do this for Presto as well.

**Reply by acvictor** [reply]
Awesome! Can I work on a PR to add these 2 fields to HiveConnectorSplit?

**Reply by majetideepak** [reply]
Yes!

**Reply by majetideepak** [reply]
Why not use `std::unordered_map<std::string, std::string> customSplitInfo;` to store file information?
Do we need the extra fields?

**Reply by majetideepak** [reply]
Or may be modify `extraFileInfo` to be a map as well.

**Reply by acvictor** [reply]
I just realized this PR checked in 3 weeks ago has added `infoColumns` https://github.com/facebookincubator/velox/pull/8800. I'll look into how modification time and file length can be added.

**Reply by majetideepak** [reply]
Good catch!

**Reply by majetideepak** [reply]
This is the Presto side change https://github.com/prestodb/presto/pull/21965/files

**Reply by acvictor** [reply]
This is a WIP, but what do you think - https://github.com/acvictor/velox/commit/4ac49e1eec13d617a3f4fc17066324caee45c966?
